### PR TITLE
tools: Add Arch Linux packaging files

### PR DESCRIPTION
**Pull request by jelly**

### CODE REVIEW

**Comment by jelly** [comment]
One issue is that shadow fails:
```
Nov 03 11:48:55 archlinux sh[241]: user 'builder': directory '/home/builder' does not exist
Nov 03 11:48:55 archlinux sh[241]: pwck: no changes
```

File: tools/arch/cockpit-wsinstance.sysuser.conf
Comment (martinpitt) [inline tools/arch/cockpit-wsinstance.sysuser.conf:1]: Is there some Arch policy that they have to be separate files? The systemd-sysusers file format allows multiple lines, and these certainly belong together.

Comment (jelly) [reply tools/arch/cockpit-wsinstance.sysuser.conf:1]: There is no policy for those files, but I guess the maintainer found it more clear to have a separate file per user.

File: test/verify/check-system-services
Comment (martinpitt) [inline test/verify/check-system-services:227]: Incidentally, @KKoukiou just cleaned that up in PR #16545 . Funny how things play out, I said "let's remove it, although it won't get in the way anytime soon".. and here we are!

Comment (jelly) [reply test/verify/check-system-services]: Cool, I rebased and dropped it.

File: test/common/testlib.py
Comment (martinpitt) [inline test/common/testlib.py:1026]: This is not quite right -- you need to strip off the distro version ("-1") as well. I independently did that this morning in https://github.com/cockpit-project/cockpit/pull/16662 as this is blocking image refreshes. After that lands, you can remove this commit.

**Comment by martinpitt** [comment]
Looking at the [TestAccounts.testUnprivileged failure](https://logs.cockpit-project.org/logs/pull-16547-20211117-164214-28aaf144-arch-bots-2619/log.html#294-2): I can't reproduce this locally with just a naïve
```
test/image-prepare -q arch
bots/vm-run arch
TEST_OS=arch test/verify/check-users TestAccounts.testUnprivileged $RUNC -stv
```
So either this got fixed in the meantime (the recent image refresh?) or some earlier test has a lasting side effect on the VM.

I also tried all 5 tests in

    TEST_OS=arch test/common/run-tests --test-dir=test/verify -n TestAccounts

This fails testBasic due to `self.assertEqual(shell, '/bin/true')` (shell is /bin/bash), but that's rather obvious; CI [fails the same way](https://logs.cockpit-project.org/logs/pull-16547-20211202-120251-a012745c-arch/log.html#252)

Then `testRootLogin` fails with a page load timeout, again same as in CI. After that it's just stuck for me, it doesn't get any further.

And indeed I can reproduce this without the run-tests driver just fine:
```
TEST_OS=arch test/verify/check-users TestAccounts.testBasic $RUNC
# this fails as above
TEST_OS=arch test/verify/check-users TestAccounts.testUnprivileged $RUNC
# this fails with "permission denied", and hangs
```
So testBasic is destroying the VM.

File: test/verify/check-static-login
Comment (martinpitt) [inline test/verify/check-static-login:436]: Similarly to the previous comment on an earlier commit, using `assertIn(..., [401, 403])` seems perfectly fine here. Also, these two changes should be in the same commit.

File: test/verify/check-storage-resize
Comment (martinpitt) [inline test/verify/check-storage-resize:163]: I don't like `|| true` in tests, it makes bugs too hard to see/debug. Please rather use `(! mountpoint {fs_dev} || umount {fs_dev}) && ...` so that we don't cover up a *failed* unmount.

File: test/verify/check-metrics
Comment (martinpitt) [inline test/verify/check-metrics:519]: missing TODO: - let's keep these grep'able

**Review by martinpitt — CHANGES_REQUESTED** [review]
Thanks! A bunch of small cleanups, but let's land this ASAP.

File: test/verify/check-pages
Comment (jelly) [reply test/verify/check-pages:129]: sorry, that's my editor which does autopep8

File: tools/arch/PKGBUILD
Comment (jelly) [reply tools/arch/PKGBUILD:10]: Ok, noted.

File: tools/arch/PKGBUILD
Comment (jelly) [reply tools/arch/PKGBUILD:28]: Not sure, it comes from the upstream packaging, we can just drop it.

File: tools/arch/PKGBUILD
Comment (jelly) [reply tools/arch/PKGBUILD:63]: `ccockpit-ws` is indeed a sysuser so it's uid is not known upfront, so that's likely why it's like it. Shall we convert these comments into an upstream ticket?

File: tools/arch/PKGBUILD
Comment (martinpitt) [reply tools/arch/PKGBUILD:28]: No, please keep the PKGBUILD exactly as it is downstream -- these were indeed comments that apply to arch downstream, and should be handled there. Of course, it is right if we temporarily try to drop these hacks and see what breaks (yay CI :grin: ), and then fix it *properly* upstream.

Comment (martinpitt) [reply tools/arch/PKGBUILD:28]: In other words, these were just unrelated comments, these obviously don't block this PR.

File: tools/arch/PKGBUILD
Comment (martinpitt) [reply tools/arch/PKGBUILD:63]: With my Debian and now Fedora packaging background this looks really strange -- but I am not familiar with Arch packaging guidelines/best practices -- perhaps that just is the sanctioned approach. So I'll leave that up to you.

File: test/verify/check-connection
Comment (jelly) [reply test/verify/check-connection:713]: Ok, splitting and fixing.

File: test/verify/check-packagekit
Comment (jelly) [reply test/verify/check-packagekit:585]: Arch actually has tracer as a package and it is added on the image, there was some issue with the test.

File: test/verify/check-storage-nfs
Comment (martinpitt) [inline test/verify/check-storage-nfs]: Rebase error -- this belongs into the first commit.

**Review by martinpitt — CHANGES_REQUESTED** [review]
Thanks! I triggered 3 tests before the review.

There's just two tiny issues left. But mainly, this is intrusive enough so that we have to run the full test suite. Please drop no-test, rebase against main, and let the tests run everywhere.

File: test/verify/check-metrics
Comment (jelly) [reply test/verify/check-metrics:691]: This is how it is:
```
[jelle@t14s][~/projects/cockpit]%pacman -Ql packagekit| grep packagekitd
packagekit /usr/lib/packagekitd
```

**Comment by martinpitt** [comment]
I dropped no-test, and force-pushed a rebase (no changes), to run all tests.

**Review by martinpitt — DISMISSED** [review]
Thanks! LGTM now, let's land when green.

**Comment by martinpitt** [comment]
This [breaks all apt package tests](https://logs.cockpit-project.org/logs/pull-16547-20211208-185123-b8bf7bd3-ubuntu-2004/log.html#287-2)

File: test/verify/packagelib.py
Comment (martinpitt) [inline test/verify/packagelib.py:378]: This needs to be "elif", otherwise the "else" branch (yum/dnf) is run for apt.

Comment (martinpitt) [reply test/verify/packagelib.py]: Fixed.

**Review by martinpitt — APPROVED** [review]

**Pull request merged**
